com.chartboost.mediation/Android~/chartboost-mediation-android-bridge/src/main/java/com/chartboost/mediation/unity/EventProcessor.kt

```diff
@@ -13,7 +15,35 @@ object EventProcessor {
         = eventConsumer.accept(placementName)
 
     @JvmStatic
-    fun serializeEventWithError(placementName: String, error: ChartboostMediationAdException?, eventConsumer: EventWithErrorConsumer<String, String>)
+    fun serializeLoadEvent(
+        placementName: String, loadId: String?, data: Map<String, String>?, error: ChartboostMediationError?,
+        loadConsumer: LoadEventConsumer<String, String, String, String, Double, String>) {
+        val errorMessage = error?.toString() ?: ""
+        var partnerId = ""
+        var auctionId = ""
+        var price =  0.0
+        val adLoadId = loadId ?: ""
+
+        data?.let { winningBidInfo ->
+            partnerId = winningBidInfo["partner_id"] ?: ""
+            auctionId = winningBidInfo["auction-id"] ?: ""
+            price = try {
+                winningBidInfo["price"]?.toDouble() ?: 0.0
```

Comment:
Nit: Instead of directly retrieving stuff from `winningBidInfo` like this, we can consider having a generalized method to do that for any Map passed in.

```kt
private fun getAuctionData(key: String, data: Map<String, String>?, defaultValue: String = ""): String {
    return data?.get(key) ?: defaultValue
}

fun serializeLoadEvent(
    placementName: String, 
    loadId: String?, 
    data: Map<String, String>?, 
    error: ChartboostMediationError?,
    loadConsumer: LoadEventConsumer<String, String, String, String, Double, String>
) {
    // ...

    val partnerId = getAuctionData("partner_id", data)
    val auctionId = getAuctionData("auction-id", data)

    // ...
}
```

Reply:
Yeah, that is a good approach, however for the new fullscreen API we are utilizing a different approach to finding data from Native. Look at AndroidExtensionMethods.cs